storage/storage-mq/src/api/amqp/PipelineConfigConsumer.ts

```diff
@@ -1,77 +1,47 @@
-import * as AMQP from "amqplib"
-import PipelineConfigEventHandler from "../pipelineConfigEventHandler"
-
-const amqpUrl = process.env.AMQP_URL!
-const pipelineConfigExchange = process.env.AMQP_PIPELINE_CONFIG_EXCHANGE!
-const pipelineConfigQueue = process.env.AMQP_PIPELINE_CONFIG_QUEUE!
-const pipelineConfigTopic = process.env.AMQP_PIPELINE_CONFIG_TOPIC!
-const pipelineConfigCreatedTopic = process.env.AMQP_PIPELINE_CONFIG_CREATED_TOPIC!
-const pipelineConfigDeletedTopic = process.env.AMQP_PIPELINE_CONFIG_DELETED_TOPIC!
+import * as AMQP from 'amqplib'
+import AmqpConsumer from '../../util/amqpConsumer'
+import PipelineConfigEventHandler from '../pipelineConfigEventHandler'
+import {
+  AMQP_URL,
+  AMQP_PIPELINE_CONFIG_EXCHANGE,
+  AMQP_PIPELINE_CONFIG_TOPIC,
+  AMQP_PIPELINE_CONFIG_QUEUE,
+  AMQP_PIPELINE_CONFIG_CREATED_TOPIC,
+  AMQP_PIPELINE_CONFIG_DELETED_TOPIC
+} from '../../env'
 
 export class PipelineConfigConsumer {
-
-    pipelineConfigEventHandler: PipelineConfigEventHandler
-
-    constructor(pipelineConfigEventHandler: PipelineConfigEventHandler) {
-        this.pipelineConfigEventHandler = pipelineConfigEventHandler
-    }
-
-    /**
-     * Connects to Amqp Service and initializes a channel
-     * @param retries   Number of retries to connect to the notification-config db
-     * @param backoff   Time to wait until the next retry
-     */
-    public async connect(retries: number, backoff: number): Promise<void> {
-        console.log("AMQP URL: " + amqpUrl)
-        for (let i = 1; i <= retries; i++) {
-          try {
-            const connection = await AMQP.connect(amqpUrl)
-            await this.initChannel(connection)
-            return
-          } catch(error) {
-            console.info(`Error connecting to RabbitMQ: ${error}. Retrying in ${backoff} seconds`)
-            console.info(`Connecting to Amqp handler (${i}/${retries})`);
-            await this.sleep(backoff)
-            continue
-          }
-        }
-        Promise.reject(`Could not establish connection to AMQP Broker (${amqpUrl})`)
-    }
-
-    private sleep(ms: number): Promise<void> {
-        return new Promise(resolve => setTimeout(resolve, ms));
+  private consumer: AmqpConsumer
+  private pipelineConfigEventHandler: PipelineConfigEventHandler
+
+  constructor (pipelineConfigEventHandler: PipelineConfigEventHandler, consumer: AmqpConsumer) {
+    this.pipelineConfigEventHandler = pipelineConfigEventHandler
+    this.consumer = consumer
+  }
+
+  async init (retries: number, msBackoff: number): Promise<void> {
+    await this.consumer.init(AMQP_URL, retries, msBackoff)
+    return this.consumer.consume(
+      AMQP_PIPELINE_CONFIG_EXCHANGE,
+      AMQP_PIPELINE_CONFIG_TOPIC,
+      AMQP_PIPELINE_CONFIG_QUEUE,
+      this.consumeEvent
+    )
+  }
+
+  // use the f = () => {} syntax to access this
```

Comment:
Why?
If you add a comment you should also provide the reason why the arrow syntax should be used.